### ISSUE TITLE
Gamestage difficulty addition rather than replacing the current difficulty

### ISSUE THREAD

**Issue by iiDogez** [SilentChaos512/ScalingHealth#352]
## Version

- 1.12.2

## What Should Be Added or Changed?

- It would be great if instead of replacing the current difficulty when you hit a game stage it would add on difficulty. For instance if I have 10 difficulty from game stage 1 and I hit game stage 2(20 difficulty) instead of just replacing the difficulty to 20 it would add 20 difficulty, making it 30 instead

## Why Should This Be Done?

- This should be added as it allows more flexibility in the difficulty system, the more milestones(game stages) the player hits the more difficult it would get, if lets say we have 8 gamestages( magic 1, magic 2, magic 3, magic 4, tech 1, tech 2, tech 3, and tech 4) and each gamestage would replace 10 difficulty more on each stage(ie magic 1 is 10 difficulty when hit, tech 3 is 30 difficulty when hit) if the player hit both tech 4 and magic 4 they would likely be a lot more powerful than the mobs at those stages, making the difficulty way too easy, as they have end game magic and tech since both gamestages are at 40 difficulty, if I were to change the difficulty system to be harder to compensate the difficulty might get too hard for those who didnt do any tech and went for magic 4 and vice versa. 
In short, it would allow for the system to be more flexible when presented with mutliple progression lines like magic vs tech
sorry for the rambling, I didnt know how to explain it very well
